sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldNotBeIgnored.cs

```diff
@@ -1,142 +1,174 @@
-        public void Foo2()
-        {
-        }
-
-        [Test]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-        [WorkItem(1234)]
-        public void Foo3()
-        {
-        }
-
-        [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-//       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant
-        public void Foo4(string s)
-        {
-        }
-
-        [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore] // Reenable when something
-        public void Foo5(string s)
-        {
-        }
-
-        [TestCase("")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-        [WorkItem(1234)]
-        public void Foo6(string s)
-        {
-        }
-
-        [TestCaseSource("DivideCases")]
-        [Microsoft.VisualStudio.TestTools.UnitTesting.Ignore]
-        // Noncompliant@-1
-        public void DivideTest(int n, int d, int q)
-        {
-            Assert.AreEqual(q, n / d);
-        }
-
-        static object[] DivideCases =
-        {
-            new object[] { 12, 3, 4 },
-            new object[] { 12, 2, 6 },
-            new object[] { 12, 4, 3 }
-        };
-    }
-}
-
-namespace Tests.Diagnostics3
-{
-    using Xunit;
-
-    class XUnitClass
-    {
-        [Fact]
-        [Ignore]
-//       ^^^^^^ Noncompliant
-        public void Foo1()
-        {
-        }
-
-        [Fact]
-        [Ignore] // Reenable when something
-        public void Foo2()
-        {
-        }
-
-        [Fact]
-        [Ignore]
-        [WorkItem(1234)]
-        public void Foo3()
-        {
-        }
-
-        [Theory]
-        [InlineData("")]
-        [Ignore]
-//       ^^^^^^ Noncompliant
-        public void Foo4(string s)
-        {
-        }
-
-        [Theory]
-        [InlineData("")]
-        [Ignore] // Reenable when something
-        public void Foo5(string s)
-        {
-        }
-
-        [Theory]
-        [InlineData("")]
-        [Ignore]
-        [WorkItem(1234)]
-        public void Foo6(string s)
-        {
-        }
-    }
-}
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Tests.Diagnostics1
+{
+    class MsTestClass
+    {
+        [TestMethod]
+        [Ignore]
+//       ^^^^^^ Noncompliant {{Either remove this 'Ignore' attribute or add an explanation about why this test is ignored.}}
+        public void Foo1()
+        {
+        }
+
+        [TestMethod]
+        [Ignore] // This test is ignored because 'blah blah'
+        public void Foo2()
+        {
+        }
+
+        [Ignore, TestMethod] // This test is ignored because 'blah blah'
+        public void Foo3()
+        {
+        }
+
+        [TestMethod]
+        [Ignore("Ignored because reasons")]
+        public void Foo4()
+        {
+        }
+
+        [TestMethod]
+        [Ignore]
+        [WorkItem(1234)]
+        public void Foo5()
+        {
+        }
+    }
+
+    [Ignore, TestClass]
+//   ^^^^^^

```

Comment:
Should this line have _Noncompliant_ on it/

Reply:
Good catch

Reply:
If you have `// ^^^` the `// Noncompliant` comment is not required; you could also include the `{{Message}}` check.

Reply:
@duncanp-sonar @Evangelink let me know if you think I should add the Noncompliant comment

Reply:
Thanks. I didn't know whether it was required or not.
I think the issues are easier to see if the Noncompliant comment is there, and it seems to be normal practice to include it.